File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.cpp
#include <string>
#include <vector>
#include <unordered_set>
using namespace std;

vector<int> solution(vector<int> arr, int k) {
    vector<int> answer;
    unordered_set<int> set;
    int temp = k;
    for(auto elem : arr)
    {
        auto it = set.find(elem);
        if(it == set.end())
        {
            set.insert(elem);
            answer.push_back(elem);
            temp--;
        }
        
        if(temp == 0)
            break;
    }
    if(answer.size() != k)
    {
        for(int i = answer.size(); i < k; i++)
            answer.push_back(-1);
    }
    return answer;
}

File: 프로그래머스/0/181858. 무작위로 K개의 수 뽑기/무작위로 K개의 수 뽑기.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int[] arr, int k) {
        List<int> answer = new List<int>();
        HashSet<int> set = new HashSet<int>();
        int temp = k;
        foreach(int elem in arr)
        {
            if(!set.Contains(elem))
            {
                set.Add(elem);
                answer.Add(elem);
                temp--;
            }
            if(temp == 0)
                break;
        }
        if(answer.Count != k)
        {
            for(int i = answer.Count; i < k; i++)
                answer.Add(-1);
        }
        return answer.ToArray();
    }
}